sudoku.go

```diff
@@ -0,0 +1,103 @@
+package sudoku
+
+import "fmt"
+
+// Solve is the main task of this app
+func Solve(puzzle []int) ([]int, error) {
```

Comment:
a better name would of been solver (we understand the document given had the function called solve())